### PR TITLE
Task/28 - Validator of full name

### PR DESCRIPTION
**Pull request by Zalezny**
To use this validator you have to add to your code:
val fullNameEditText = findViewById(R.id.full_name_edit_text)
FullNameValidator.attach(fullNameEditText, this)

### CODE REVIEW

File: app/src/main/java/com/safetyheads/akademiaandroida/utils/FullNameVaildator.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        } else if (fullName.isEmpty()) {
+            editText.error = context.getString(R.string.invalid_empty_message)
+            return
+
+        } else if (spacesAmount > 1) {
+            editText.error = context.getString(R.string.invalid_spaces_message)
+        } else {
+            editText.error = null
+            return
+
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/utils/FullNameVaildator.kt:66]: remove empty line

**Review by przemek19980102 — CHANGES_REQUESTED** [review]
Some minor changes here. Could you also provide a description on how to test the change?

File: app/src/main/java/com/safetyheads/akademiaandroida/utils/FullNameVaildator.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import com.safetyheads.akademiaandroida.R
+
+/*
+* To use this validator you have to add to your code:
+* FullNameValidator.attach(fullNameEditText, this)
+*/
+
+object FullNameValidator {
+
+    private const val maxTextLength = 10
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/utils/FullNameVaildator.kt:16]: It should be 100, not 10

File: app/src/main/java/com/safetyheads/akademiaandroida/utils/FullNameVaildator.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import android.widget.EditText
+import androidx.core.widget.addTextChangedListener
+import com.safetyheads.akademiaandroida.R
+
+/*
+* To use this validator you have to add to your code:
+* FullNameValidator.attach(fullNameEditText, this)
+*/
+
+object FullNameValidator {
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/utils/FullNameVaildator.kt:14]: Something is not right, when I type "ABC" then it returns that surname is ok even if there is no surname

**Review by przemek19980102 — CHANGES_REQUESTED** [review]

**Review by przemek19980102 — APPROVED** [review]

**Review by msarelo — DISMISSED** [review]

**Review by przemek19980102 — APPROVED** [review]

**Review by msarelo — APPROVED** [review]

**Pull request merged**